config/v1/types_cluster_version.go

```diff
@@ -95,6 +95,13 @@ type ClusterVersionStatus struct {
 	// +required
 	Desired Release `json:"desired"`
 
+	// architecture indicates whether the cluster is a homogeneous or
```

Comment:
If this is determined live from inspection of nodes, why just single and multi instead of a list of architectures with numbers of nodes?  

If this is not determined live from inspection of nodes, why is this a status field instead of a spec field where a user states their intent?

Reply:
It's determined by inspection of the payload at load time with this change https://issues.redhat.com/browse/ART-4129. And per https://coreos.slack.com/archives/CB95J6R4N/p1657141996473319 ART will only include the new metadata setting `multi` if it's a heterogeneous arch, i.e. ART does not set an arch for a homogeneous payload in which case CVO pulls it via runtime.GOARCH.

Reply:
Since this value can change (see the Trevor's response above), this is looking like a spec field where the desired state is set and a status field where the actual current state of the cluster is reflected.  That way it becomes possible to manage a transition.